### PR TITLE
(SIMP-8613) appropriately configure firewalld

### PR DESCRIPTION
**Pull request by trevor-vaughan**
Fixed:
  * Ensure that firewalld is only enabled on systems that have it
    installed.

SIMP-8613 #close

### CODE REVIEW

**Review by op-ct — APPROVED** [review]
:eye: 👍

File: manifests/init.pp [hunk 1]
```diff
@@ -120,7 +120,7 @@
   simplib::assert_metadata($module_name)
 
   if $enable != 'ignore' {
-    if $use_firewalld {
+    if ( 'firewalld' in pick($facts['simplib__firewalls'], 'none') ) and $use_firewalld {
```
Comment (lnemsick-simp) [inline manifests/init.pp:123]: shouldn't this be `[ 'none' ]`

Comment (trevor-vaughan) [reply manifests/init.pp:123]: @lnemsick-simp It'll return either the Array from `$facts['simplib__firewalls']` or the String `none`.

Comment (trevor-vaughan) [reply manifests/init.pp:123]: @lnemsick-simp Ah...well, I was wrong. It does indeed pick the first entry from the Array :-/.

This means that some other code needs to be fixed as well.

Comment (trevor-vaughan) [reply manifests/init.pp:123]: @lnemsick-simp OK, I had a bad test. `pick()` does work as expected in this case.

**Comment by lnemsick-simp** [comment]
I was wrong.  `pick` is being used correctly.  The docs are confusing but the implementation of the function is quite clear.  Code is good as is.

**Comment by lnemsick-simp** [comment]
Please fix the documentation at the beginning of init.pp and regenerate the REFERENCE.md.  The discussion on how to enable firewalld mode is incorrect.  `simp_options::firewall` is a Boolean and cannot be set to `firewalld`.

**Review by lnemsick-simp — APPROVED** [review]

**Pull request merged**